locale/inc/leatherman/locale/locale.hpp

```diff
@@ -95,63 +95,197 @@ namespace leatherman { namespace locale {
      * @param domain The catalog domain to use for i18n via gettext.
      * @return The translated string.
      */
-    std::string translate_c(std::string const& context, std::string const& single, std::string const& plural, int n, std::string const& domain = PROJECT_NAME);
+    std::string translate_np(std::string const& context, std::string const& single, std::string const& plural, int n, std::string const& domain = PROJECT_NAME);
+
+    /**
+     * Translates and formats text using the locale initialized by this library.
+     * Use the default domain, i.e. PROJECT_NAME.
+     * Replaces the use of boost::format with a variadic function call.
+     * Specialized to the PROJECT_NAME domain.
+     * @param s The format string.
+     * @param args Format arguments.
+     * @return The string generated by translating the format string, then applying the arguments.
+     */
+    template <typename... TArgs>
+    std::string format(std::string const& s, TArgs... args)
+    {
+        // Create and apply formatter here, as we want to guarantee the lifetime of the arguments.
+        // boost::locale::format doesn't make copies, and a common gotcha is using temporary arguments
+        // to build up the formatter.
+        // Technique for the one-liner explained at http://florianjw.de/en/variadic_templates.html.
+#ifdef LEATHERMAN_I18N
+        static const std::string domain{PROJECT_NAME};
```

Comment:
Yes, I was bothered by the replication as well. Let me take a look at your lambda-based approach and alternatives.